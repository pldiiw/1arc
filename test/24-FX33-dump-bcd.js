'use strict';

let engine = require('engine').initialize();
const dumpBCD = require('instructions').dumpBCD;

engine.data[0xB] = 237;
engine.data[0xE] = 14;
engine.I = 100;

const engine1 = dumpBCD(engine, 0xB);
const engine2 = dumpBCD(engine1, 0xE);

console.log(engine1.memory[100]);
console.log(engine1.memory[101]);
console.log(engine1.memory[102]);
console.log(engine1.I);

console.log(engine2.memory[100]);
console.log(engine2.memory[101]);
console.log(engine2.memory[102]);
console.log(engine2.I);
