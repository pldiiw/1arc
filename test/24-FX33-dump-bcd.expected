2
3
7
100
0
1
4
100
